src/main/ts/config.ts

```diff
@@ -154,7 +158,9 @@ export const resolveConfig = (pluginConfig: TAnyMap, context: TContext, path = P
     branch: opts.branch || DEFAULT_BRANCH,
     enterprise,
     token,
-    repo
+    repo,
+    currentBranch: opts.currentBranch || DEFAULT_CURRENT_BRANCH,
```

Comment:
Yes! I think this is a good improvement to keep things compact. I will also add the logger piece and the util function you mentioned then commit that change to this PR as well in a WIP commit.

Reply:
And one more thing. I suppose, we should inherit `branch` value from semrel config, but now it's omitted by `resolveOptions`. This may be tweaked smth like:

```typescript
export const resolveOptions = (pluginConfig...)
...
return { ...base, ...extra, _branch:  pluginConfig.branch}
```

```typescript
const pullTagsBranch = anyDefined(opts.pullTagsBranch, opts._branch)
```

Reply:
Is this always available? If not then something like 
`const pullTagsBranch = anyDefined(opts.pullTagsBranch, opts._branch, DEFAULT_PULL_TAGS_BRANCH)`
Would make more sense.

Reply:
`semrel` should assign it, afair. Anyway, one more fallback is ok.